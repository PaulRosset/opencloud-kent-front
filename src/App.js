import React, { Component } from "react";
import Header from "./components/Header";
import Body from "./components/Body";

class App extends Component {
  render() {
    return [<Header key="1" />, <Body key="2" />];
  }
}

export default App;
