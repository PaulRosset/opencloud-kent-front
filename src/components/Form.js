import React, { Component } from "react";
import agent from "superagent";
import styled from "styled-components";
import { Icon } from "semantic-ui-react";
import Dropzone from "react-dropzone";
import Portal from "./Advert";

const DropZoned = styled(Dropzone)`
  position: relative;
  border: 10px dotted #545454;
  border-radius: 20px;
  color: #000;
  font: bold 24px/200px arial;
  height: 200px;
  margin: 30px auto;
  text-align: center;
  width: 100%;
`;

const Advert = styled.p`
  font-size: 15px;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  margin: auto;
  width: 100%;
  text-align: center;
`;

export default class FormUpload extends Component {
  constructor(props) {
    super(props);
    this.state = {
      openPortal: false
    };
  }

  handleOpen() {
    this.setState({
      openPortal: true
    });
  }

  handleClose() {
    this.setState({
      openPortal: false
    });
  }

  onDrop(acceptedFiles, rejectedFiles) {
    if (acceptedFiles[0]) {
      this.handleOpen();
      agent
        .post("http://l'urlquetuveuxjeanclaude.com")
        .use(() => this.setState({ loading: true }))
        .attach(acceptedFiles[0].name, acceptedFiles[0])
        .end((err, res) => {
          if (!err) {
            this.setState({
              header: `${acceptedFiles[0].name} book has been uploaded !`,
              message: "Processing the book ...",
              color: "green",
              loading: false
            });
          }
        });
    } else if (rejectedFiles) {
      this.handleOpen();
      this.setState({
        header: `${rejectedFiles[0].name} book can't be uploaded`,
        message: `Please verify the validity format of the book, expected pdf got ${
          rejectedFiles[0].type
        }`,
        color: "red"
      });
    }
  }

  render() {
    return (
      <div>
        <DropZoned
          onDrop={(acceptedFiles, rejectedFiles) =>
            this.onDrop(acceptedFiles, rejectedFiles)
          }
          accept="application/pdf"
        >
          <Advert>
            Drop your book (pdf) file right here, we take care of the rest{" "}
            <span role="img" aria-label="emoji happy">
              ðŸ˜€
            </span>
            <br />
            <Icon
              name="upload"
              size="big"
              color="grey"
              link
              bordered
              circular
              style={{ marginTop: "30px" }}
            />
          </Advert>
        </DropZoned>
        <Portal
          open={this.state.openPortal}
          handleOpen={this.handleOpen.bind(this)}
          handleClose={this.handleClose.bind(this)}
          header={this.state.header}
          message={this.state.message}
          color={this.state.color}
          loading={this.state.loading}
        />
      </div>
    );
  }
}
