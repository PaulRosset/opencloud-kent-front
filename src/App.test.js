import React from "react";
import ReactDOM from "react-dom";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";

import App from "./App";
import Form from "./components/Form";
import Dropzone from "react-dropzone";
import { setTimeout } from "timers";

Enzyme.configure({ adapter: new Adapter() });

let wrapperForm;

describe("Test <Form /> Component", () => {
  beforeAll(() => {
    wrapperForm = shallow(<Form />);
  });

  it("Check initial <Form /> Component", () => {
    expect(wrapperForm.getElement()).toMatchSnapshot();
  });

  it("Check after change some state to <Portal /> Component", () => {
    wrapperForm.setState({
      loading: true,
      color: "#000",
      header: "Test Header",
      message: "Message Test"
    });
    expect(wrapperForm.getElement()).toMatchSnapshot();
  });
});

describe("Test <App /> Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
  });
});
