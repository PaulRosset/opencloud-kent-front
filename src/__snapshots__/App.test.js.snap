// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Form /> Component Check after change some state to <Portal /> Component 1`] = `
<div
  className="form-container"
>
  <Styled(Dropzone)
    accept="application/pdf"
    onDrop={[Function]}
  >
    <styled.p>
      Drop your book (pdf) file right here, we take care of the rest
       
      <span
        aria-label="emoji happy"
        role="img"
      >
        ðŸ˜€
      </span>
      <br />
      <Icon
        as="i"
        bordered={true}
        circular={true}
        color="grey"
        link={true}
        name="upload"
        size="big"
        style={
          Object {
            "marginTop": "30px",
          }
        }
      />
    </styled.p>
  </Styled(Dropzone)>
  <Portal
    color="#000"
    handleClose={[Function]}
    handleOpen={[Function]}
    header="Test Header"
    loading={true}
    message="Message Test"
    open={false}
  />
</div>
`;

exports[`Test <Form /> Component Check initial <Form /> Component 1`] = `
<div
  className="form-container"
>
  <Styled(Dropzone)
    accept="application/pdf"
    onDrop={[Function]}
  >
    <styled.p>
      Drop your book (pdf) file right here, we take care of the rest
       
      <span
        aria-label="emoji happy"
        role="img"
      >
        ðŸ˜€
      </span>
      <br />
      <Icon
        as="i"
        bordered={true}
        circular={true}
        color="grey"
        link={true}
        name="upload"
        size="big"
        style={
          Object {
            "marginTop": "30px",
          }
        }
      />
    </styled.p>
  </Styled(Dropzone)>
  <Portal
    color={undefined}
    handleClose={[Function]}
    handleOpen={[Function]}
    header={undefined}
    loading={undefined}
    message={undefined}
    open={false}
  />
</div>
`;
